### ISSUE TITLE
"Protocol Filters" is not working

### ISSUE THREAD

**Issue by cheerfultime** [Cacti/plugin_flowview#118]
Hi,

AlmaLinux release 9.4 + cacti + flowview(3.3 latest develop from github on Jun 5)
Ubuntu 22.04 + cacti + flowview(3.3 latest develop from github on Jun 5)

1. "Protocol Filters" are not working
I entered the IP in "Source IP" but it's not working.
Is my setting wrong?
![ProtocolFilters](https://github.com/Cacti/plugin_flowview/assets/55739637/45fc5991-8a1b-4c41-b236-7f7145772293)

2. Then, there are error messages in cacti.log.
2024-06-06 14:26:18 - ERROR PHP WARNING in  Plugin 'flowview': Undefined array key "src_domain" in file: /usr/share/cacti/plugins/flowview/functions.php  on line: 16542024-06-06 14:26:18 - CMDPHP PHP ERROR WARNING Backtrace:  (/plugins/flowview/flowview.php[51]:flowview_get_chartdata(), /plugins/flowview/functions.php[1654]:CactiErrorHandler())

Thanks!

**Comment by TheWitness** [comment]
I've updated the functions.php.  Please update and try again.  In the future if there is such an issue, there will be a warning in your Cacti.log.  I will need to know the report you were running in this case.

**Comment by cheerfultime** [comment]
> I've updated the functions.php. Please update and try again. In the future if there is such an issue, there will be a warning in your Cacti.log. I will need to know the report you were running in this case.

Hi,

1.
"Protocol Filters" are not working < = there is no error messages in cacti.log.
It's still not working.

2.
After i update flowview, restart flow-capture, disable flowview, uninstall flowview.
Then install flowview, enable flowview, there are error messages in cacti.log.

2024-06-06 22:52:21 - FLOWVIEW ERROR: Unable to process domain information.  Please open a ticket on GitHub
2024-06-06 22:52:21 - FLOWVIEW ERROR: Unable to process domain information.  Please open a ticket on GitHub
2024-06-06 22:52:21 - FLOWVIEW ERROR: Unable to process domain information.  Please open a ticket on GitHub

3.
I found that as long as "uninstall flowview",  all the plugin_flowview_raw_2024xxx tab will be deleted.
May I ask if this is normal?

Thanks!

**Comment by TheWitness** [comment]
1. You will have to be more specific.
2. That's log message is what I was looking for.
3. That's normal.  Don't uninstall unless you are prepared to loose that data.  In a few hours, you will get them back again.  Restore from your last backup if you need the data back.

**Comment by TheWitness** [comment]
Should be resolved now.  I reproduced and corrected the reports.

**Comment by cheerfultime** [comment]
Hi,

I have updated flowview.

1.
![firefox_AzxCkXUOw5](https://github.com/Cacti/plugin_flowview/assets/55739637/fd2044da-4806-46cc-9e2b-83073d3fc73f)
"Source IP" or "Dest IP" in "Protocol Filters" works for IP, but not for range(x.x.x.x/24).
When fill range in "Source IP" or "Dest IP", there is no error messages in cacti.log.

2.
![firefox_t1a7pV7ISq](https://github.com/Cacti/plugin_flowview/assets/55739637/60c61d37-ed22-407d-bf3a-6f84ea5bfd98)
When "Statistical Report" choose "Destination IP", the table has an additional "Source DNS" field.

3.
![firefox_WwxPG3gw9D](https://github.com/Cacti/plugin_flowview/assets/55739637/f8ce35f4-fa6b-4049-9e94-31054350b7d9)
Since I unchecked "Domains/Hostnames Only" once, there are error messages in cacti.log.

2024-06-08 00:47:14 - CMDPHP Validation Error, Variable:graph_type, Value:null Backtrace:  (/plugins/flowview/flowview.php[52]:flowview_request_vars(), /plugins/flowview/flowview.php[280]:validate_store_request_vars(), /lib/html_utility.php[712]:die_html_input_error(), /lib/html_validate.php[64]:cacti_debug_backtrace())
2024-06-08 00:47:14 - CMDPHP Validation Error, Variable:graph_type, Value:null Backtrace:  (/plugins/flowview/flowview.php[52]:flowview_request_vars(), /plugins/flowview/flowview.php[280]:validate_store_request_vars(), /lib/html_utility.php[712]:die_html_input_error(), /lib/html_validate.php[64]:cacti_debug_backtrace())
2024-06-08 00:47:14 - CMDPHP Validation Error, Variable:graph_type, Value:null Backtrace:  (/plugins/flowview/flowview.php[52]:flowview_request_vars(), /plugins/flowview/flowview.php[280]:validate_store_request_vars(), /lib/html_utility.php[712]:die_html_input_error(), /lib/html_validate.php[64]:cacti_debug_backtrace())


Thanks!

**Comment by TheWitness** [comment]
Good update.  That validation error should be transient.

**Comment by TheWitness** [comment]
Just did another update.  Hopefully this one gets it.

**Comment by cheerfultime** [comment]
Hi,

I have updated flowview.

1.
"Source IP" or "Dest IP" in "Protocol Filters" works for IP, but still not for range(x.x.x.x/24).
Is there anything I should pay attention to when entering "Source IP" or "Dest IP" range?
When fill range in "Source IP" or "Dest IP", there is no error messages in cacti.log.

2.
![Destination IP](https://github.com/Cacti/plugin_flowview/assets/55739637/dc046192-5b3a-41e6-a672-d40cfff0fa56)
When "Statistical Report" choose "Destination IP", the table has additional "Source DNS" and "Source IP" field.
Are those specially added fields?

![Source IP](https://github.com/Cacti/plugin_flowview/assets/55739637/751ec6e8-c557-4417-a438-1ede20f32d65)
When "Statistical Report" choose "Source IP", "Dest DNS" and "Dest IP" will not be displayed.

3.
"Validation Error" problem has been solved, thank you.

4.
Is there an upper limit on the efficiency of FlowView's collection of flows?
I want to collect flows in the scope of B class, but I find that the flow data is missing and incomplete.
There is no error messages in cacti.log.
The CPU usage of the server is very low, there is more than 24GB of ram,  more than 40GB of hard disk space, and 10Gb network card interface.
Do I need to adjust php or mariadb or other config?


Thanks!
